pkgs/tools/misc/qmk/default.nix

```
@@ -27,13 +39,23 @@ buildPythonApplication rec {
     appdirs
     argcomplete
     colorama
-    dotty-dict
+    #dotty-dict # See https://pypi.org/project/qmk-dotty-dict/
+    qmk-dotty-dict
     hid
     hjson
     jsonschema
     milc
     pygments
+    pyrsistent
     pyusb
+  ] ++ [ # Binaries need to be in the path so this is in propagatedBuildInputs
+    avrdude
+    dfu-programmer
+    dfu-util
```

```suggestion
    dfu-util
    teensy-loader-cli
```
Please also add `teensy-loader-cli`, so that Teensy boards can be flashed (the tool is tiny and does not have any dependencies except libusb, which is already in the dependency tree).